Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -83,6 +83,15 @@ extension HTTPClientResponse {
         @inlinable public func makeAsyncIterator() -> AsyncIterator {
             .init(storage: self.storage.makeAsyncIterator())
         }
+
+        /// Accumulates an ``Swift/AsyncSequence`` of ``ByteBuffer``s into a single ``ByteBuffer``.
+        /// - Parameters:
+        ///   - maxBytes: The maximum number of bytes this method is allowed to accumulate
+        /// - Throws: `NIOTooManyBytesError` if the the sequence contains more than `maxBytes`.
+        /// - Returns: the number of bytes collected over time
+        func collect(maxBytes: Int) async throws -> ByteBuffer {
+            return try await self.collect(upTo: maxBytes)
```

Comment:
If the response already announces its body size we can check that size before downloading anything. This is e.g. the case if the response contains a `Content-Length` header which contains the expected body size. If it is bigger than `maxBytes` we should throw before calling `self.collection(upTo:)`. To actually do that we need to move this extension from `HTTPClientResponse.Body` to  `HTTPClientResponse` or have some kind of reference in `HTTPClientResponse.Body` to the headers.

Be careful though, there are edge cases where we don't expect a body even if `Content-Length` is set. This is e.g. the case if the request was just a `HEAD` request. I think there a are more but that's the only one I know of the top of my head.

Reply:
I do not think this is appropriate for helper functions to check the `Content-Length`, but will confirm that it deals with it correctly.